### PR TITLE
adding algorithms that parse through react fiber tree data

### PR DESCRIPTION
**Pull request by eileenlee115**
Adding algorithms that parse through react fiber tree data

- getNodes is a function that recursively gets all nodes from the react fiber tree
- displayComponentTree is a function that displays a rough outline of the parent-child-sibling relationships among our components using d3
- addDepth is a function that adds a depth property onto our node object
- getNamedComponents is a function that grabs only the named components

Screen Shot 2021-06-15 at 8 18 22 PM

### CODE REVIEW

**Review by brian-z-liang — APPROVED** [review]

File: webpack.react.config.js [hunk 1]
```diff
@@ -1,21 +1,21 @@
-const path = require("path");
```
Comment (sturgeonphillip) [inline webpack.react.config.js:1]: need to update to es modules

File: src/atoms.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  default: {width: '600px',
+  height: '900px'}
+})
+
+atoms.reactState = atom({
+  key: 'reactState',
+  default: [],
+})
+
+export default atoms
```
Comment (sturgeonphillip) [inline src/atoms.js]: add line

File: src/StateDisplay.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+
+return (
+  <>
+  <h1> State Data</h1>
+  
+  </>
+)
+}
+export default StateDisplay
```
Comment (sturgeonphillip) [inline src/StateDisplay.tsx]: add line

**Review by sturgeonphillip — APPROVED** [review]
Great work!

**Pull request merged**